### PR TITLE
Scrap updates

### PR DESCRIPTION
**Pull request by AztronautApe**
implemented update announc
newly generated requirements.txt

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -57,26 +59,34 @@ def _parseString(self,message,target):
...
+                    fileContent = json.load(file)
+                    latestArticle = scrapper.CheckCurrentNewsArticle("http://maplestory2.nexon.net/en/news")
+                    fileContent['URL'] = latestArticle
+                    latestArticleDate = scrapper.GetNewsArticleDate()
+                    fileContent['publishedOn'] = latestArticleDate
+                if fileContent['URL'] == "":
+                    #Add a news article if nothing was given
+                    with open(ms2NewsJSON, "w") as write_file:
+                        json.dump(fileContent, write_file, indent=4)
+                elif fileContent['URL'] != scrapper.CheckCurrentNewsArticle("http://maplestory2.nexon.net/en/news"):
```
Comment (DMunkei) [inline bot.py]: We already have the variable stored  in latestArticle dont call the function again to make the check.

File: bot.py [hunk 2]
```diff
@@ -57,26 +59,34 @@ def _parseString(self,message,target):
 
     async def _postNewNewsArticle(self):
         """Writes into the updates channel once a new news article is posted."""        
-        scrapper = Scrapper()
+        scrapper = Scrapper.Scrapper()
```
Comment (DMunkei) [inline bot.py]: Why Scrapper.Scrapper() ? Me don't understand :D

File: bot.py [hunk 3]
```diff
@@ -57,26 +59,34 @@ def _parseString(self,message,target):
...
+                elif fileContent['URL'] != scrapper.CheckCurrentNewsArticle("http://maplestory2.nexon.net/en/news"):
+                    #if the latest news article is not already in json
+                    fileContent['URL'] = latestArticle
+                    fileContent['publishedOn'] = latestArticleDate
+                    with open(ms2NewsJSON, "w") as write_file:
+                        json.dump(fileContent, write_file, indent=4)
+                        await updateChannel.send("http://maplestory2.nexon.net" + latestArticle)
+            except Exception as e:
+                print(e)
+        await asyncio.sleep(360)  # task runs every 60 seconds
```
Comment (DMunkei) [inline bot.py]: change into 3600. That checks every minute not hour.

File: Scrapper.py [hunk 4]
```diff
@@ -14,11 +14,12 @@ def __init__(self):
         def GetWebsite(self, targetURL):
                 self.url = targetURL
                 self.httpResponse = urllib.request.Request(self.url,data=None,headers={"User-Agent":
-                "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.92 Safari/537.36"})
+                "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36"})
```
Comment (DMunkei) [inline Scrapper.py]: I'm not sure if using a different user agent will make problems or not.  Just change it back to what it was before.

"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.92 Safari/537.36"})

File: bot.py (hunk 2 above)
Comment (AztronautApe) [reply bot.py]: Scrapper is a module, containing the class Scrapper
you don't want to instantiate a module you want to instatiate a class right?

File: bot.py (hunk 1 above)
Comment (AztronautApe) [reply bot.py]: what variable?

File: bot.py (hunk 2 above)
Comment (DMunkei) [reply bot.py]: Ah okay. Yes sir 👍

File: bot.py (hunk 1 above)
Comment (DMunkei) [reply bot.py]: Check line 72.

File: bot.py (hunk 3 above)
Comment (AztronautApe) [reply bot.py]: whoops

**Pull request merged**